pkg/analysis/enums/testdata/src/a/a.go

```diff
@@ -0,0 +1,141 @@
+package a
+
+// Valid: +kubebuilder:validation:Enum without explicit values (auto-discovers constants)
+// +kubebuilder:validation:Enum=Pending;Running;Succeeded;Failed
+type Phase string
+
+const (
+	PhasePending   Phase = "Pending"   // Valid PascalCase
+	PhaseRunning   Phase = "Running"   // Valid PascalCase
+	PhaseSucceeded Phase = "Succeeded" // Valid PascalCase
+	PhaseFailed    Phase = "Failed"    // Valid PascalCase
+)
+
+// Valid: +k8s:enum always auto-discovers constants
+// +k8s:enum
+type State string
+
+const (
+	StateActive   State = "Active"   // Valid PascalCase
+	StateInactive State = "Inactive" // Valid PascalCase
+)
+
+// Valid: +kubebuilder:validation:Enum with explicit values (doesn't check constants)
+// +kubebuilder:validation:Enum=Pending;helper
+type ExplicitPhase string
+
+// These constants are helpers - not validated because marker has explicit values
+const (
+	ExplicitPending ExplicitPhase = "Pending"
+	explicit_helper ExplicitPhase = "helper" // Not flagged - explicit values mode
+)
+
+// Invalid: Type alias without enum marker (will be flagged when used in fields)
+type Status string
+
+const (
+	StatusReady    Status = "Ready"
+	StatusNotReady Status = "NotReady"
+)
+
+// Invalid: Type with kubebuilder:validation:Enum marker but underlying type is not string
+// +kubebuilder:validation:Enum
+type InvalidEnumType int // want "type InvalidEnumType has enum marker but underlying type is not string"
+
+// Invalid enum values (not PascalCase) - using auto-discovery
+// +kubebuilder:validation:Enum
+type BadPhase string
+
+const (
+	phase_pending BadPhase = "pending"      // want "enum value \"pending\" should be PascalCase"
+	phase_failed  BadPhase = "Phase-Failed" // want "enum value \"Phase-Failed\" should be PascalCase"
+)
+
+// Valid: Acronyms and all-caps are allowed
+// +kubebuilder:validation:Enum
+type Protocol string
+
+const (
+	ProtocolHTTP  Protocol = "HTTP"  // Valid: acronym
+	ProtocolHTTPS Protocol = "HTTPS" // Valid: acronym
+	ProtocolTCP   Protocol = "TCP"   // Valid: acronym
+)
+
+// Test struct with fields
+type MySpec struct {
+	// Valid: uses type alias with enum marker
+	Phase Phase
+
+	// Valid: uses type alias with enum marker
+	State State
+
+	// Valid: plain string (not required to be enum unless RequireTypeAliasForEnums is enabled)
+	PlainString string
+
+	// Invalid: type alias without enum marker
+	Status Status // want "field MySpec.Status uses type Status which appears to be an enum but is missing an enum marker \\(\\+kubebuilder:validation:Enum or \\+k8s:enum\\)"
```

Comment:
`+enum` is the basic case, we should indicate `+enum` as the suggested marker for now